# Brain Guide

> â›” MANDATORY workflow for Archon Brain canvas system. AI MUST follow EXACTLY or canvas will NOT work.

---

## ğŸš« STOP! READ BEFORE ANY FILE OPERATIONS

**DO NOT CREATE ANY FILES until you have:**
1. âœ… Read and understood the Blocking Rules below
2. âœ… Calculated the exact `fileName` from canvas name
3. âœ… Verified ALL folder names will match `fileName`
4. âœ… Planned node positions using Grid Layout (no overlap!)

**If you skip these steps, the canvas WILL BE BROKEN and show "0 nodes" in Archon app.**

---

## â›” BLOCKING RULE #1: fileName (ZERO TOLERANCE)

**If canvas `name:` value and folder names don't match, Archon app CANNOT load nodes!**

### fileName Generation Algorithm (Archon App Internal Logic)

```
1. Replace spaces with hyphens: "My Canvas" â†’ "My-Canvas"
2. Convert to lowercase: "My-Canvas" â†’ "my-canvas"
3. Keep only letters, numbers, hyphens: "Pattern: V2" â†’ "pattern-v2"
```

**Transformation Examples:**

| `name:` value | Generated fileName |
|---------------|-------------------|
| `"My Canvas"` | `my-canvas` |
| `"Aesthetic Canvas: Minimal Diary"` | `aesthetic-canvas-minimal-diary` |
| `"UI Design (v2)"` | `ui-design-v2` |

### Folder Naming Rule (MUST MATCH fileName)

```
Canvas file:    {fileName}.md
Nodes folder:   {fileName}_Nodes/
Connections:    {fileName}_Connections/
Datasheet:      {fileName}_Datasheet/
```

### âœ… CORRECT Example

```yaml
# Canvas file: aesthetic-minimal-diary.md
name: "Aesthetic Minimal Diary"  # fileName = aesthetic-minimal-diary
```
```
Docs/Brain/
â”œâ”€â”€ aesthetic-minimal-diary.md
â”œâ”€â”€ aesthetic-minimal-diary_Nodes/      âœ… MATCH
â”œâ”€â”€ aesthetic-minimal-diary_Connections/
â””â”€â”€ aesthetic-minimal-diary_Datasheet/
```

### âŒ WRONG Example (App shows 0 nodes)

```yaml
# Canvas file: aesthetic-minimal-diary-brain.md
name: "Aesthetic Canvas: Minimal Diary Brain"
# fileName = aesthetic-canvas-minimal-diary-brain (colon removed!)
```
```
Docs/Brain/
â”œâ”€â”€ aesthetic-minimal-diary-brain.md
â”œâ”€â”€ aesthetic-minimal-diary-brain_Nodes/  âŒ MISMATCH!
â”‚   # App looks for: aesthetic-canvas-minimal-diary-brain_Nodes
```

### Pre-Creation Checklist (MANDATORY)

**Before creating ANY files:**

1. [ ] Calculate fileName from `name:` value
2. [ ] Canvas file = `{fileName}.md`
3. [ ] Nodes folder = `{fileName}_Nodes/`
4. [ ] Connections folder = `{fileName}_Connections/`
5. [ ] Datasheet folder = `{fileName}_Datasheet/`
6. [ ] **ALL folder names EXACTLY match the calculated fileName**

---

## â›” BLOCKING RULE #2: Node Positioning (NO OVERLAP)

**Nodes MUST be placed on grid. Overlapping nodes = unusable canvas.**

```
Position X = COLUMN Ã— 350
Position Y = ROW Ã— 250
```

**Before creating nodes, assign grid positions:**
```
Node 1: Col 0, Row 0 â†’ position: { x: 0, y: 0 }
Node 2: Col 1, Row 0 â†’ position: { x: 350, y: 0 }
Node 3: Col 0, Row 1 â†’ position: { x: 0, y: 250 }
Node 4: Col 1, Row 1 â†’ position: { x: 350, y: 250 }
```

---

## â›” BLOCKING RULE #3: Mandatory Validation

**After creating ALL files, AI MUST verify:**

```
âœ“ VALIDATION CHECKLIST (AI must print this)
â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”
Canvas name:     "{name value}"
Calculated fileName: "{result}"
â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”
âœ“ Canvas file:   {fileName}.md
âœ“ Nodes folder:  {fileName}_Nodes/ (contains {N} files)
âœ“ Connections:   {fileName}_Connections/ (if applicable)
âœ“ Datasheet:     {fileName}_Datasheet/ (if applicable)
â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”
Node positions verified: No overlaps
â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”
```

**If validation fails, FIX IMMEDIATELY before reporting completion.**

---

## About Brain Organizer

The Brain Organizer is a systematic workflow for using Archon's Brain canvas system. Brain canvases are visual knowledge organization tools that help structure research, references, and complex concepts.

**Key Capabilities:**
- Create and manage visual canvases for organizing information
- Support multiple node types (Heading, Text, Image, Memo, Data-Sheet)
- Group related nodes with color coding
- Connect nodes to show relationships

## When to Use (AUTOMATIC)

**âš ï¸ AI MUST automatically follow this guide when:**
- ANY file operation in `Docs/Brain/` directory
- User mentions "Brain", "canvas", "nodes", "ìº”ë²„ìŠ¤", "ë¸Œë ˆì¸"
- Creating visual research output
- Organizing aesthetic references
- Any task requiring structured information visualization

**Trigger Phrases:**
- "Create a Brain canvas for {topic}"
- "Organize {info} in Brain"
- "Add nodes to Brain canvas"
- "Brain-organize {research results}"
- "ë¸Œë ˆì¸ ìº”ë²„ìŠ¤ ë§Œë“¤ì–´ì¤˜"
- "ì •ë³´ ì •ë¦¬í•´ì„œ ë¸Œë ˆì¸ì—"

**Use Cases:**
- Aesthetic research output organization
- Visual reference collection
- Pattern library creation
- Cultural reference mapping
- Complex concept visualization

## Directory Structure

**âš ï¸ All folder names MUST use `{fileName}` calculated from canvas `name:` value!**

```
Docs/Brain/
â”œâ”€â”€ {fileName}.md                      # Main canvas file
â”œâ”€â”€ {fileName}_Nodes/                  # Node files directory
â”‚   â”œâ”€â”€ Node_Heading-Text_{UUID}.md
â”‚   â”œâ”€â”€ Node_Body-Text_{UUID}.md
â”‚   â”œâ”€â”€ Node_Post-It_{UUID}.md
â”‚   â”œâ”€â”€ Node_Image_{UUID}.md
â”‚   â”œâ”€â”€ Node_Memo_{UUID}.md
â”‚   â””â”€â”€ Node_Data-Sheet_{UUID}.md
â”œâ”€â”€ {fileName}_Connections/            # Connection files directory
â”‚   â”œâ”€â”€ Connection_Arrow_{UUID}.md
â”‚   â””â”€â”€ Connection_Normal_{UUID}.md
â””â”€â”€ {fileName}_Datasheet/              # Datasheet files directory
    â”œâ”€â”€ Datasheet_{UUID}.csv
    â””â”€â”€ {UUID}.styles.json
```

## Workflow

### Phase 1: Canvas Planning

1. **Canvas Purpose**: Determine type (aesthetic, technical, conceptual)
2. **Canvas Name**: Choose name, then calculate fileName
   - Recommended: Use kebab-case directly (e.g., `my-research`)
   - Alternative: Human-readable â†’ calculate fileName
3. **Node Types**: Plan Heading-Text, Image, Memo, Data-Sheet nodes

### Phase 2: Canvas Creation

**File**: `Docs/Brain/{fileName}.md`

**Required YAML Metadata:**
```yaml
---
id: "{UUID}"
name: "{Name}"
viewport_offset: { x: 0, y: 0 }
zoom_level: 1.0
created_at: "{ISO8601}"
updated_at: "{ISO8601}"
node_ids: []
connections: []
groups: []
---
```

### Phase 3: Node Creation

**Directory**: `Docs/Brain/{fileName}_Nodes/`

| Information Type | Node Type | File Format |
|------------------|-----------|-------------|
| Category label | Heading-Text | `Node_Heading-Text_{UUID}.md` |
| Detailed analysis | Body-Text | `Node_Body-Text_{UUID}.md` |
| Quick note | Post-It | `Node_Post-It_{UUID}.md` |
| Visual reference | Image | `Node_Image_{UUID}.md` |
| Critical observation | Memo | `Node_Memo_{UUID}.md` |
| Comparison table | Data-Sheet | `Node_Data-Sheet_{UUID}.md` |

### Phase 4: Grouping

Group 3+ related nodes with color coding:
- Red `#FF6B6B`: Primary category
- Teal `#4ECDC4`: Secondary category
- Yellow `#F7DC6F`: Highlights
- Green `#98D8C8`: Completed/validated

### Phase 5: Connections

**Directory**: `Docs/Brain/{fileName}_Connections/`

- `arrow`: Directional (cause â†’ effect)
- `normal`: Bi-directional (association)
- Positions: `top`, `bottom`, `left`, `right`

### Phase 6: Finalization

1. **Verify fileName match** (CRITICAL)
2. Update canvas metadata with all node_ids
3. Verify directory structure
4. Output summary with fileName verification

## Integration with Aesthetic Skills

| Aesthetic Skill | Brain Output |
|-----------------|--------------|
| `aesthetic-cultural-research` | Image + Memo nodes |
| `aesthetic-critic-historian` | Memo nodes |
| `aesthetic-form-composition` | Memo nodes |
| `aesthetic-motion-temporal` | Memo nodes |
| `aesthetic-pattern-miner` | Data-Sheet nodes |

## Positioning Strategy (âš ï¸ Prevent Node Overlap)

**Grid-Based Layout - Use this formula:**
```
Position X = COLUMN Ã— 350
Position Y = ROW Ã— 250
```

**Example:**
```yaml
Node 1: position: { x: 0, y: 0 }
Node 2: position: { x: 350, y: 0 }
Node 3: position: { x: 0, y: 250 }
```

## Best Practices

- Max 50 nodes per canvas for performance
- Use kebab-case for canvas names (safest approach)
- Always create nodes in `Docs/Brain/` directory
- Use proper UUID format for node identification

## Relationship to Claude Code

In Claude Code, this workflow is available as the `brain-organizer` skill which can be invoked directly. For other AI tools, follow this guide manually when organizing information in Brain canvases.

---

## ğŸš¨ FINAL REMINDER: 3 BLOCKING RULES

```
â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”“
â”ƒ RULE #1: fileName MUST match folder names                   â”ƒ
â”ƒ RULE #2: Nodes MUST use grid positions (X=COLÃ—350, Y=ROWÃ—250)â”ƒ
â”ƒ RULE #3: MUST print validation checklist after completion   â”ƒ
â”—â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”›

VIOLATION = BROKEN CANVAS (0 nodes, overlapping nodes, unusable)
```

---
*Generated by Archon*