# Spec Document Management Rules

> Zero-tolerance rules for managing Spec documents. AI MUST follow these rules strictly.

## â›” CRITICAL: Spec File Update Enforcement

**AI MUST NEVER create new Spec markdown files. AI MUST ONLY update existing template files.**

### Absolute File Path Rules

**ONLY these files are allowed for Spec updates:**

| Development Phase | File Path (ABSOLUTE, NO VARIABLES) | Status |
|-------------------|-----------------------------------|--------|
| POC | `Docs/Spec/POC_Spec.md` | Required |
| MVP | `Docs/Spec/MVP_Spec.md` | Required |
| PILOT | `Docs/Spec/PILOT_Spec.md` | Required |
| MMP | `Docs/Spec/MMP_Spec.md` | Required |
| PMF | `Docs/Spec/PMF_Spec.md` | Required |
| SCALE | `Docs/Spec/SCALE_Spec.md` | Required |
| LIVE | `Docs/Spec/LIVE_Spec.md` | Required |

### Zero Tolerance Items

âŒ **FORBIDDEN - These actions will FAIL:**
- Creating files like `POC.md`, `MVP_Specification.md`, `Spec_POC.md`, `specs.md`
- Creating files in locations other than `Docs/Spec/`
- Using variables or dynamic file names like `{phase}_Spec.md`
- Creating temporary Spec files anywhere
- Creating files with similar names like `POC-Spec.md`, `poc_spec.md`, `POCSpec.md`

âœ… **REQUIRED - AI MUST:**
- Use EXACT file paths listed in the table above
- Update existing `*_Spec.md` files in `Docs/Spec/` directory
- If file doesn't exist, create it using the exact template structure below
- Preserve existing file structure (sections, headers, tables)

---

## ðŸ“Š Phase Scope Guide: Development First Principle

**Development is ALWAYS the top priority. Operations/Marketing can only be considered from MMP onwards.**

### Phase-by-Phase Focus

| Phase | Primary Focus | Non-Dev Tasks |
|-------|---------------|---------------|
| **POC** | ðŸ”§ **100% Development** - Core technology validation | âŒ Do not include |
| **MVP** | ðŸ”§ **100% Development** - Core feature implementation | âŒ Do not include |
| **PILOT** | ðŸ”§ **100% Development** - User testing features | âŒ Do not include |
| **MMP** | ðŸ”§ **Development first** + Deployment/Operations start | âš ï¸ Can consider |
| **PMF** | ðŸ”§ **Development first** + Monitoring/Analytics | âš ï¸ Can consider |
| **SCALE** | ðŸ”§ **Development first** + Infrastructure/Marketing | âš ï¸ Can consider |
| **LIVE** | ðŸ”§ **Development first** + Operations/Community | âš ï¸ Can consider |

### What Counts as "Development"

âœ… **Development Tasks (Priority in ALL phases):**
- Feature implementation (UI, Logic, API)
- Bug fixes
- Refactoring
- Test writing
- Performance optimization
- Architecture improvements

âš ï¸ **Non-Development Tasks (Consider from MMP onwards):**
- Deployment pipeline setup
- Monitoring/Logging systems
- Marketing strategy planning
- Community management
- User documentation

### Important Distinction

**System development FOR marketing â‰  Marketing tasks**

| Request | Classification | When to Include |
|---------|----------------|-----------------|
| "Build analytics dashboard" | Development task | Anytime OK |
| "Plan marketing analytics strategy" | Non-dev task | MMP onwards |
| "Implement push notification system" | Development task | Anytime OK |
| "Push notification marketing campaign" | Non-dev task | MMP onwards |

---

## Enforcement Mechanism

**Before any Spec-related write operation, AI MUST:**

1. **Identify Phase**: Determine which development phase (POC/MVP/PILOT/MMP/PMF/SCALE/LIVE)
2. **Construct Path**: Build EXACT path: `Docs/Spec/{PHASE}_Spec.md`
3. **Verify Existence**: Check if file exists
   - If YES: Read and update existing content
   - If NO: Create file with template structure
4. **Update Only**: Modify content within existing structure

---

## Example Workflows

### âŒ WRONG:
```
User: "Write POC spec"
AI: Creates "POC_Specification.md" or "Docs/POC.md" or "specs/POC.md"
```

### âœ… CORRECT:
```
User: "Write POC spec"
AI:
1. Target path: Docs/Spec/POC_Spec.md
2. Check if file exists
3. If NO: Create with template structure
4. Update content in Docs/Spec/POC_Spec.md
```

---

## File Structure Lock

**When updating Spec files, preserve these sections:**

```markdown
# {Phase} Specification

> **Stage**: {Phase}
> **Last Modified**: {Date}
> **Status**: {Status}

---

## Summary
## 1. Goals
## 2. Scope
## 3. Technical Specifications
## 4. Modules
## 5. Deliverables
## 6. Risks & Mitigations
## 7. Notes
```

---
*Generated by Archon*